corehq/form_processor/submission_post.py

```diff
@@ -221,25 +234,34 @@ def run(self):
         if failure_response:
             return FormProcessingResult(failure_response, None, [], [], 'known_failures')
 
+        try:
+            # decode xml to ensure it can be decoded to string when and if needed in pre processing steps
+            instance_xml = self.instance.decode() if isinstance(self.instance, bytes) else self.instance
+        except UnicodeDecodeError as e:
+            return get_submission_error(self.domain, self.instance, e, self.auth_context.to_json())
```

Comment:
returning this here is an error, there is additional functionality for handling submission errors below in the `if submitted_form.is_submission_error_log:` block. This function must return a `corehq.form_processor.submission_post.FormProcessingResult` not a `corehq.form_processor.parsers.form.FormProcessingResult` (I just realized that these 2 have the same name which seems bad)

Reply:
https://github.com/dimagi/commcare-hq/pull/28161/commits/f3d78aef9343449d99439ebd1e18ff4d3041ec7e

and

renamed https://github.com/dimagi/commcare-hq/pull/28161/commits/68a5d83e33a6ecbe6852befaa59c4fabe12ad68d